File: nfcd/src/main/java/it/convergent/obliterator/xposed/Native.java
package it.convergent.obliterator.xposed;

/**
 * Created by altamic on 23/10/16.
 */

class Native {
    static {
        Instance = new Native();
    }
    public static final Native Instance;
    public native void uploadConfiguration(byte atqa, byte sak, byte[] hist, byte[] uid, byte[] data);
    public native void setEnabled(boolean enabled);
    public native boolean isEnabled();
}

File: nfcd/src/main/java/it/convergent/obliterator/xposed/Native.kt
package it.convergent.obliterator.xposed

class Native {
    companion object {
        val Instance = Native()
    }

    external fun setEnabled(value: Boolean): Void
    external fun isEnabled(): Boolean
    external fun uploadConfiguration(atqa: Byte, sak: Byte, uid: ByteArray, data: ByteArray)
}
